src/Compilers/CSharp/Portable/Symbols/PublicModel/PreprocessingSymbol.cs

```diff
@@ -12,10 +12,14 @@ namespace Microsoft.CodeAnalysis.CSharp.Symbols.PublicModel
     internal sealed class PreprocessingSymbol : IPreprocessingSymbol
     {
         private readonly string _name;
+        private readonly IAssemblySymbol _assembly;
+        private readonly IModuleSymbol _module;
```

Comment:
They enable the reference finder to find the project in which to search for the symbol. I opted for this approach because it seemed to me that there would be need for a much harder and complicated implementation for preprocessing symbols specifically. If you think it could be approached less intrusively, a suggestion would be welcome.

Reply:
where is that done?  looking at the PR it looks like you search all documents.  I don't see the usages of tehse fields at all from teh IDE side.

Reply:
The initial invocation relies on the project. I said elsewhere that it didn't seem worthy to me to specialize logic for a solution-wide implementation of the finder's execution, though I might be missing some existing code that already does this.

Reply:
> The initial invocation relies on the project.

How so?  (note: most of my questions are really asking you to flesh out these answers.  where does this happen?  why does it happen?  etc. etc.).

Reply:
I didn't dig through that; I did find that it wouldn't work without an assembly tied to the sought symbol, and realized that the finder looks through the declaring assembly's bound project. To avoid this behavioral adjustment, a solution-wide finder invocation must be implemented.

Reply:
i odn't understand the need for any of this.

Reply:
this comment still applies.  i have no idea why this is needed.